### DISCUSSION TITLE
How do you use zstd (or others) compression for parquet?

### DISCUSSION THREAD

**Discussion by theelderbeever** [delta-io/delta-rs#1464] [Q&A]
As the question states above I would like to use zstd instead of snappy for writing parquet files to delta lake using the rust crate. I haven't been able to find anything that mentions customizing this in the writer. Any clarity would be super helpful.

**Comment by wjones127** [comment]
For Rust, this is in passed in through [WriterProperties](https://docs.rs/parquet/41.0.0/parquet/file/properties/struct.WriterProperties.html).

In Python, this isn't exposed yet. It's being tracked in https://github.com/delta-io/delta-rs/issues/1465

**Reply by theelderbeever** [reply]
@wjones127 thanks for pointing me towards that. 

Looking into some of the writer code it seems like the only place that `WriterProperties` is passable is in the `PartitionWriter` [here](https://github.com/delta-io/delta-rs/blob/0dda99bbb080d5cf8a6ecc350d0a789e7fa56549/rust/src/writer/record_batch.rs#L269). Are there plans to expose this field elsewhere such as on the `RecordBatchWriter::try_new()`? From what I can tell you have to custom roll the `RecordBatchWriter{ ... }` ([here](https://github.com/delta-io/delta-rs/blob/0dda99bbb080d5cf8a6ecc350d0a789e7fa56549/rust/src/writer/record_batch.rs#L55)) to set any properties.

Or am I missing something?

**Reply by wjones127** [reply]
Oh I think you are right. I though it would be exposed on the write operation here:

https://github.com/delta-io/delta-rs/blob/0dda99bbb080d5cf8a6ecc350d0a789e7fa56549/rust/src/operations/write.rs

But it isn't 😢

**Reply by theelderbeever** [reply]
Is this worth turning into an issue?

**Reply by wjones127** [reply]
Yes, feel free to file a new issue for that.
